tests/test_cache.py

```diff
@@ -765,6 +761,23 @@ def on_get(key, value, existed):
     assert log == "miss=None, existed=False"
 
 
+def test_cache_on_set(cache: Cache):
+    """Test that on_set(cache) callback."""
+    log = ""
+
+    def on_set(key, new_value, old_value):
+        nonlocal log
+        log = f"{key}={new_value}, old_value={old_value}"
+
+    cache.on_set = on_set
+
+    cache.set("a", 1)
+    assert re.match(r"^a=1, old_value=<object object at 0x.*>$", log)
```

Comment:
I think we can simplify the assertion by not using `re.match`. One suggestion would be to make `log` a dictionary and set it to something like `log = {"key": key, "new_value": new_value, "old_value": old_value}`. Then can do `assert log == {"key": "a", "new_value": 1, "old_value": UNSET}`.